packages/smooth_app/lib/cards/product_cards/knowledge_panels/knowledge_panel_title_card.dart

```diff
@@ -20,12 +20,14 @@ class KnowledgePanelTitleCard extends StatelessWidget {
     Color? colorFromEvaluation;
     if (evaluation != null &&
         (knowledgePanelTitleElement.iconColorFromEvaluation ?? false)) {
-      colorFromEvaluation = _getColorFromEvaluation(evaluation!);
+      if (themeData.brightness == Brightness.dark) {
+        colorFromEvaluation = _getColorFromEvaluationDarkMode(evaluation!);
+      } else {
+        colorFromEvaluation = _getColorFromEvaluation(evaluation!);
+      }
     }
-    if (evaluation != null &&
-        _getColorFromEvaluation(evaluation!) == null &&
-        themeData.brightness == Brightness.dark) {
-      colorFromEvaluation = Colors.white;
+    if (evaluation != null && themeData.brightness == Brightness.dark) {
+      colorFromEvaluation = _getColorFromEvaluationDarkMode(evaluation!);
```

Comment:
I don't see the point of this. You more or less already dealt with that case a bit earlier.

Reply:
Yes, you are correct. I've removed this code.